rules/import-style.js

```diff
@@ -0,0 +1,320 @@
+	// An exotic custom parser or a bug in one could cover it too.
+	/* istanbul ignore next */
+	return [];
+};
+
+const joinOr = words => {
+	return words
+		.map((word, index) => {
+			if (index === (words.length - 1)) {
+				return word;
+			}
+
+			if (index === (words.length - 2)) {
+				return word + ' or';
+			}
+
+			return word + ',';
+		})
+		.join(' ');
+};
+
+const MESSAGE_ID = 'importStyle';
+
+// Keep this alphabetically sorted for easier maintenance
+const defaultStyles = {
+	chalk: {
+		default: true
+	},
+	path: {
+		default: true
+	},
+	util: {
+		named: true
+	}
+};
+
+const templates = eslintTemplateVisitor({
+	parserOptions: {
+		sourceType: 'module',
+		ecmaVersion: 2018
+	}
+});
+
+const variableDeclarationVariable = templates.variableDeclarationVariable();
+const assignmentTargetVariable = templates.variable();
+const moduleNameVariable = templates.variable();
+
+const assignedDynamicImportTemplate = templates.template`async () => {
+	${variableDeclarationVariable} ${assignmentTargetVariable} = await import(${moduleNameVariable});
+}`.narrow('BlockStatement > :has(AwaitExpression)');
+
+const assignedRequireTemplate = templates.template`
+	${variableDeclarationVariable} ${assignmentTargetVariable} = require(${moduleNameVariable});
+`;
+
+const create = context => {
+	let [
+		{
+			styles = {},
+			extendDefaultStyles = true,
+			checkImport = true,
+			checkDynamicImport = true,
+			checkRequire = true
+		} = {}
+	] = context.options;
+
+	styles = extendDefaultStyles ?
+		defaultsDeep({}, styles, defaultStyles) :
+		styles;
+
+	styles = new Map(
+		Object.entries(styles).map(
+			([moduleName, styles]) =>
+				[moduleName, new Map(Object.entries(styles))]
+		)
+	);
+
+	const report = (node, moduleName, actualImportStyles, allowedImportStyles, isRequire = false) => {
+		if (!allowedImportStyles || allowedImportStyles.size === 0) {
+			return;
+		}
+
+		let effectiveAllowedImportStyles = allowedImportStyles;
+
+		if (isRequire) {
+			// For `require`, `'default'` style allows both `x = require('x')` (`'namespace'` style) and
+			// `{default: x} = require('x')` (`'default'` style) since we don't know in advance
+			// whether `'x'` is a compiled ES6 module (with `default` key) or a CommonJS module and `require`
+			// does not provide any automatic interop for this, so the user may have to use either of theese.
+			if (allowedImportStyles.get('default') && !allowedImportStyles.get('namespace')) {
+				effectiveAllowedImportStyles = new Map(allowedImportStyles);
+				effectiveAllowedImportStyles.set('namespace', true);
+			}
+		}
+
+		if (actualImportStyles.every(style => effectiveAllowedImportStyles.get(style))) {
+			return;
+		}
+
+		const data = {
+			allowedStyles: joinOr([...allowedImportStyles.keys()]),
+			moduleName
+		};
+
+		context.report({
+			node,
+			messageId: MESSAGE_ID,
+			data
+		});
+	};
+
+	return templates.visitor({
+		ImportDeclaration(node) {
+			if (!checkImport) {
+				return;
+			}
```

Comment:
Can we add `listener`, and don't add `ImportDeclaration` at all if `checkImport === false`